nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestRemoveRecordField.java

```diff
@@ -0,0 +1,396 @@
+import org.apache.nifi.schema.inference.SchemaInferenceUtil;
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class TestRemoveRecordField {
+
+    private TestRunner runner;
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    @Before
+    public void setup() throws InitializationException {
+        runner = TestRunners.newTestRunner(RemoveRecordField.class);
+
+        final JsonRecordSetWriter jsonWriter = new JsonRecordSetWriter();
+        runner.addControllerService("writer", jsonWriter);
+        runner.setProperty(jsonWriter, SchemaAccessUtils.SCHEMA_ACCESS_STRATEGY, SchemaAccessUtils.INHERIT_RECORD_SCHEMA);
+        runner.setProperty(jsonWriter, "suppress-nulls", "never-suppress");
+        runner.setProperty(jsonWriter, "Pretty Print JSON", "true");
+        runner.setProperty(jsonWriter, "Schema Write Strategy", "full-schema-attribute");
+        runner.enableControllerService(jsonWriter);
+        runner.setProperty(AbstractRecordProcessor.RECORD_WRITER, "writer");
+    }
+
+    @Test
+    public void testNotRootPath() throws InitializationException, IOException {
+        setUpJsonReader(null);
+
+        runner.setProperty("root_path", "/");
+        runner.assertNotValid();
+    }
+
+    @Test
+    public void testNotRootPathEL() throws InitializationException, IOException {
+        // must have at least 1 dynamic property to be valid
+        runner.assertNotValid();
+
+        setUpJsonReader(null);
+
+        runner.setProperty("root_path", "${remove.path}");
+        runner.assertValid();
+
+        runner.enqueue(
+                "{}".getBytes(StandardCharsets.UTF_8),
+                Collections.singletonMap("remove.path", "/")
+        );
+
+        runner.run();
+        runner.assertAllFlowFilesTransferred(AbstractRecordProcessor.REL_FAILURE, 1);
+        runner.getFlowFilesForRelationship(AbstractRecordProcessor.REL_FAILURE).get(0)
+                .assertAttributeEquals("record.error.message", "org.apache.nifi.processor.exception.ProcessException Thrown");
+    }
+
+    @Test
+    public void testRemoveSimpleFieldWithSchemaInference() throws InitializationException, IOException {
+        final String inputContent = "src/test/resources/TestRemoveRecordField/input/complex-person.json";
+        final String outputContent = "src/test/resources/TestRemoveRecordField/output/complex-person-no-dateOfBirth.json";
```

Comment:
The numerous paths should remove the `src/test/resources/TestRemoveRecordField` repeated prefix using a shared utility method.